### PR TITLE
feat(Message): add isPartial() method

### PR DESCRIPTION
**Pull request by fasterplayer**
**Please describe the changes this PR makes and why it should be merged:**
This is to providde an easier method to identify whether or not a message is Partial for Typescript Users

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Comment by PuneetGopinath** [comment]
I think title should be `feat(Message): add isPartial() method`

**Comment by fasterplayer** [comment]
> I think title should be `feat(Message): add isPartial() method`

Thanks, changed it.

**Comment by PuneetGopinath** [comment]
Also you added
```js
public isPartial(): this is PartialMessage;
```
Shouldn't that be
```js
public isPartial(): boolen;
```

File: src/structures/Message.js [hunk 1]
```diff
@@ -809,6 +809,14 @@ class Message extends Base {
     return this.edit({ attachments: [] });
   }
 
+  /**
+   * Indicates whether this Message is Partial.
```
Comment (DaStormer) [inline src/structures/Message.js:813]:
```suggestion
   * Indicates whether this message is a partial.
```

**Comment by fasterplayer** [comment]
> Also you added
> 
> ```js
> public isPartial(): this is PartialMessage;
> ```
> 
> Shouldn't that be
> 
> ```js
> public isPartial(): boolen;
> ```

No, the point of adding this Method is that using this method is that Typescript will know if the message is a PartialMessage.
But the method will still return a boolean.

File: src/structures/Message.js [hunk 2]
```diff
@@ -809,6 +809,14 @@ class Message extends Base {
     return this.edit({ attachments: [] });
   }
 
+  /**
+   * Indicates whether this message is a partial.
```
Comment (vladfrangu) [inline src/structures/Message.js:813]:
```suggestion
   * Indicates whether this message is partial.
```

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

File: src/structures/Message.js (hunk 2 above)
Comment (DaStormer) [reply src/structures/Message.js]: There are a lot of occurrences that say "a partial" instead of just "partial." Should those be changed? 🤔

**Comment by wasdennnoch** [comment]
I don't see why this is needed. The typings for the full `message` already say that `partial` is always `false`:
https://github.com/discordjs/discord.js/blob/75b48d8d0f70338c75721df3dbd8fc37866383bc/typings/index.d.ts#L1121
And `Partialize` makes sure that for `PartialMessage` it's always `true`:
https://github.com/discordjs/discord.js/blob/75b48d8d0f70338c75721df3dbd8fc37866383bc/typings/index.d.ts#L4296

TS can infer the type of the message based on that:

**Comment by fasterplayer** [comment]
> I don't see why this is needed. The typings for the full `message` already say that `partial` is always `false`:
> https://github.com/discordjs/discord.js/blob/75b48d8d0f70338c75721df3dbd8fc37866383bc/typings/index.d.ts#L1121
> 
> 
> And `Partialize` makes sure that for `PartialMessage` it's always `true`:
> https://github.com/discordjs/discord.js/blob/75b48d8d0f70338c75721df3dbd8fc37866383bc/typings/index.d.ts#L4296
> 
> TS can infer the type of the message based on that:
> 



> I don't see why this is needed. The typings for the full `message` already say that `partial` is always `false`:
> https://github.com/discordjs/discord.js/blob/75b48d8d0f70338c75721df3dbd8fc37866383bc/typings/index.d.ts#L1121
> 
> 
> And `Partialize` makes sure that for `PartialMessage` it's always `true`:
> https://github.com/discordjs/discord.js/blob/75b48d8d0f70338c75721df3dbd8fc37866383bc/typings/index.d.ts#L4296
> 
> TS can infer the type of the message based on that:
> 
> 
> The only situation in which I can imagine `isPartial()` to be useful is when you type something to be of type `Message`, and then can branch if `isPartial()` says that isn't actually a `PartialMessage`. But then the issue isn't that the Discord.JS typings are wrong - it's that your own app's typings are wrong, which say that an object is always of type `Message` when in fact it isn't.

After making a new test, it seems like you are write. I really don't know how i went through this issue. I think that my issue was only because i was trying to check if message was an instance of Message and it was returning true when it was partial. Really sorry for the time wasted. This PR is actually useless. I'm closing it.

**Pull request closed without merging**